.github/workflows/review.yml

```diff
@@ -6,31 +6,68 @@ on:
   workflow_dispatch:
 
 jobs:
-  load-build-group:
-    name: Load Build Group Config JSON
+  build:
+    name: Build for all supported Node versions
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: ['14', '16', '18']
+    
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v3
+
+      - name: Setup Node ${{ matrix.node }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: 'npm'
+
+      - name: Show Environment Info
+        run: |
+          printenv
+          node --version
+          npm --version
+          cat /etc/os-release
+
+      - name: NPM Install dependencies
+        run: npm install
+
+      # runs: node build.js for the current node version
+      - name: Build from source for Node ${{ matrix.node }}
+        run: npm run build
+
+      - name: Upload built packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: npm-${{ matrix.node }}
+          path: npm/*
+          retention-days: 1
+
+  load-group:
+    name: Load Prebuilt Group Config JSON
     runs-on: ubuntu-latest
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
 
     steps:
       - name: Checkout ${{ github.ref }}
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-        # build with the lowest versions of the OSs supported.
-        # ensures the glibc/musl versions are the oldest/most compatible.
-      - name: Load build group data
+        # contains images that are unable to build "out of the box" and will install prebuilt
+      - name: Load prebuilt group data
```

Comment:
Minor nit: with this new approach we are supporting all platforms with a prebuilt, right? Can consider a name change to just treat these as the "tested group".

Reply:
Yeah, I've ended up moving everything to a single tested group in the oboe 12 PR coming after this